### DISCUSSION TITLE
Dynamic rewrites via vercel.json

### DISCUSSION THREAD

**Discussion by floriangosse** [vercel/community#1054] [Help] [answered]
I want to utilize the rewrite functionality to proxy an API endpoint but the endpoint has to be different for the environments.

For now I've done with a middleware and an environment variable but would like to use the environment inside of the rewrite rules.

**Comment by amyegan** [comment, marked as answer]
It sounds like you're already using the solution I would recommend, @floriangosse! [Middleware](https://vercel.com/docs/concepts/functions/edge-middleware#) is letting you use environment variables for rewrites, while a `vercel.json` config is not able to access them.

**Comment by floriangosse** [comment]
Thanks for your quick answer. Would have been nice to use the rewrite feature because it reduces the usage functions.

Maybe it could be a useful solution for the future to support environment variables in some parts of the `vercel.json`.
